include/z64.h

```diff
@@ -777,16 +777,19 @@ typedef struct {
     /* 0x08 */ DmaRequest  dmaRequest;
     /* 0x28 */ OSMesgQueue loadQueue;
     /* 0x40 */ OSMesg   loadMsg;
-} ObjectStatus; // size = 0x44
+} ObjectLoadEntry; // size = 0x44
+
+// 1000 kilobytes
+#define OBJECT_SPACE_SIZE_BASE (1000 * 1024)
 
 typedef struct {
     /* 0x0000 */ void*  spaceStart;
     /* 0x0004 */ void*  spaceEnd; // original name: "endSegment"
-    /* 0x0008 */ u8     num; // number of objects in bank
-    /* 0x0009 */ u8     unk_09;
-    /* 0x000A */ u8     mainKeepIndex; // "gameplay_keep" index in bank
-    /* 0x000B */ u8     subKeepIndex; // "gameplay_field_keep" or "gameplay_dangeon_keep" index in bank
-    /* 0x000C */ ObjectStatus status[OBJECT_EXCHANGE_BANK_MAX];
+    /* 0x0008 */ u8     loadEntriesCount; // total amount of used load entries
+    /* 0x0009 */ u8     persistentLoadEntriesCount; // amount of load entries that won't be reused when loading a new object list (when loading a new room)
+    /* 0x000A */ u8     mainKeepIndex; // "gameplay_keep" index in loadEntries
+    /* 0x000B */ u8     subKeepIndex; // "gameplay_field_keep" or "gameplay_dangeon_keep" index in loadEntries
+    /* 0x000C */ ObjectLoadEntry loadEntries[19];
```

Comment:
I went the even crazier route of using "entry"/"object entry" everywhere, it actually works out quite well. no more bank/load discussion

Reply:
I'm not a fan of using that everywhere personally. I think it's good to use the term entry for the struct in the context, but we lose a lot of information by not having "index" in the other names. This is especially apparent for things like `objectEntry = Object_GetEntry(&play->objectCtx, actorInit->objectId);`, which are everywhere and are specifically intended to get the index which can then be used to get info on the object. I would agree with the name if the function returned a pointer to the entry ofc, but that's not the case so it's confusing imo.

That being said, I don't think "index" or "entry index" alone is enough, because it's not just the index in the list of object entries, it's a fixed value that won't change in the current room, so actors often store it once and continue using it over their lifetime. It's almost an ID in that sense, but we've been avoiding that in order to not get it confused with the actual object ID.

All this is why I think the term "bank index" fits the best among everything we've mentioned so far. It's recognizable enough that you can quickly associate it with this pattern and its usage as an index, while not being too easy to mistake with the actual "object id".

It could be nice to find an even better alternative that fits all these boxes tho, but I don't have any ideas myself atm

Reply:
I think the only reason "bank" would be better than "entry" is because it's what has been used so far / what comes from the debug string, there doesn't seem to be anything else to it

Furthermore the only way I see "bank" making sense is the "(computing) A contiguous block of memory that is of fixed, hardware-dependent size, but often larger than a page and partitioning the memory such that two distinct banks do not overlap." sense ( https://en.wiktionary.org/wiki/bank#Noun_3 ("bank" is also a super overloaded word)) which is somewhat accurate (besides the load stuff and the entries being only a pointer to the "bank", not the "bank" itself), but doesn't really have anything to do with the engine's objects in particular

"entry" is more generic which makes it counter intuitively more appropriate than "bank" which as a word is overloaded

I think it's ok to use `entry` as an index without saying `entryIndex`. If you want I can change the few `entry` pointers to `entryPtr`, that would be less verbose than changing everything else to `entryIndex`

Reply:
I am usually a hater of the word "index" being added to things, but I think it does matter here. just `entry` makes it seem like the entry itself, it just doesnt feel right